collector/queries.go

```diff
@@ -0,0 +1,415 @@
+		LEFT JOIN (
+			SELECT workspace_id, job_id, name
+			FROM system.lakeflow.jobs
+			WHERE delete_time IS NULL
+			QUALIFY ROW_NUMBER() OVER (PARTITION BY workspace_id, job_id ORDER BY change_time DESC) = 1
+		) j ON t.workspace_id = j.workspace_id AND t.job_id = j.job_id
+		WHERE t.period_start_time >= current_timestamp() - INTERVAL 2 HOURS
+			AND t.result_state IS NOT NULL
+		GROUP BY t.workspace_id, t.job_id, j.name, t.result_state
+	`
+
+	// jobRunDurationQuery calculates duration quantiles (p50, p95, p99) per job.
+	// using Databricks' percentile_approx function for server-side aggregation.
+	// This reduces memory footprint compared to client-side calculation.
+	// JOINs with system.lakeflow.jobs to get job names.
+	jobRunDurationQuery = `
+		SELECT 
+			workspace_id,
+			job_id,
+			job_name,
+			percentile_approx(duration_seconds, 0.5) as p50,
+			percentile_approx(duration_seconds, 0.95) as p95,
+			percentile_approx(duration_seconds, 0.99) as p99
+		FROM (
+			SELECT 
+				t.workspace_id,
+				t.job_id,
+				COALESCE(j.name, 'unknown') as job_name,
+				unix_timestamp(t.period_end_time) - unix_timestamp(t.period_start_time) as duration_seconds
+			FROM system.lakeflow.job_run_timeline t
+			LEFT JOIN (
+				SELECT workspace_id, job_id, name
+				FROM system.lakeflow.jobs
+				WHERE delete_time IS NULL
+				QUALIFY ROW_NUMBER() OVER (PARTITION BY workspace_id, job_id ORDER BY change_time DESC) = 1
+			) j ON t.workspace_id = j.workspace_id AND t.job_id = j.job_id
+			WHERE t.period_start_time >= current_timestamp() - INTERVAL 2 HOURS
+				AND t.period_end_time IS NOT NULL
+				AND t.period_end_time > t.period_start_time
+		)
+		GROUP BY workspace_id, job_id, job_name
+	`
+
+	// taskRetriesQuery counts retries across job tasks from the task timeline table.
+	// A task retry is detected when the same task_key within a job_run_id runs multiple times.
+	// The difference between total task runs and unique (job_run_id, task_key) pairs = retry count.
+	// JOINs with system.lakeflow.jobs to get job names.
+	taskRetriesQuery = `
+		SELECT 
+			t.workspace_id,
+			t.job_id,
+			COALESCE(j.name, 'unknown') as job_name,
+			t.task_key,
+			COUNT(*) - COUNT(DISTINCT CONCAT(t.job_run_id, '-', t.task_key)) as retry_count
+		FROM system.lakeflow.job_task_run_timeline t
+		LEFT JOIN (
+			SELECT workspace_id, job_id, name
+			FROM system.lakeflow.jobs
+			WHERE delete_time IS NULL
+			QUALIFY ROW_NUMBER() OVER (PARTITION BY workspace_id, job_id ORDER BY change_time DESC) = 1
+		) j ON t.workspace_id = j.workspace_id AND t.job_id = j.job_id
+		WHERE t.period_start_time >= current_timestamp() - INTERVAL 2 HOURS
+			AND t.job_run_id IS NOT NULL
+		GROUP BY t.workspace_id, t.job_id, j.name, t.task_key
+		HAVING COUNT(*) > COUNT(DISTINCT CONCAT(t.job_run_id, '-', t.task_key))
+	`
+
+	// jobSLAMissQuery identifies jobs exceeding a configured SLA threshold.
+	// Currently hardcoded to 3600 seconds (1 hour) - should be configurable.
```

Comment:
If we want it to be configurable, then please provide it as a default value in a variable, ideally one that you can configure in a command-line flag or config file :)